Assets/Scripts/Controllers/Web3Controller.cs

```diff
@@ -26,29 +44,12 @@ private void Start()
             Destroy(gameObject);
             return;
         }
-
-#if UNITY_EDITOR
-        // Use the local Hardhat test network
-        provider = new RPCWeb3Provider()
-        {
-            ChainId = 31337,
-            PrivateKey = "0xac0974bec39a17e36ba4a6b4d238ff944bacb478cbed5efcae784d7bf4f2ff80",
-            Network = "Hardhat",
-            RpcUrl = "http://localhost:8545"
-        };
-#elif UNITY_WEBGL
-        GameObject webglProvider = new GameObject("WebGL_Web3Provider");
-        provider = webglProvider.AddComponent<WebGLWeb3Provider>();
-#else
-        provider = null;
-#endif
     }
 
-    public async void ConnectWallet()
+    public void ConnectWallet()
     {
         Debug.Log("Trying to connect");
-        string addr = await provider.ConnectToWallet(new Web3ConnectArgs { chainId = defaultChainId });
-        Debug.Log("Got addr: " + addr);
+        MetamaskConnectButton_Clicked();
         onWalletConnected?.Invoke();
```

Comment:
The wallet won't actually be connected at this point, so this event should be called in `EthereumEnabled()` instead.

Reply:
I think we can call it immediately if this is a DEBUG build though